### PR TITLE
Add packaging tools

### PR DESCRIPTION
**Pull request by acmiyaguchi**
There are a lot of changes in here, but I had to modify the project shape in order to package this properly. I want to deploy this to the Databricks clusters, which are running [Ubuntu 16.04 LTS](https://docs.databricks.com/release-notes/runtime/4.3.html#system-environment).

The notable changes:
* the generated shared object is now prefixed with `_libprio` instead of `_prio`
* the generated bindings are in `prio/libprio.py` instead of `prio/lib/prio.py`
* `from prio.lib import prio` is now `from prio import libprio`
* a `MANIFEST.in` gathers `mprio.h`, `mprio.a`, and `mpi.a` from the libprio submodule

I've uploaded the package to the [testing instance of pypi](https://test.pypi.org/project/prio/). 
```bash
$ pipenv sync --dev && pipenv shell
$ make
$ python setup.py sdist
$ twine upload --repository-url https://test.pypi.org/legacy/ dist/prio-0.2.tar.gz
```

I tested this with a local docker container that contained the examples.

```bash
$ docker build -t . prio
$ docker run -it prio bash
$ pip3 install --index-url https://test.pypi.org/simple/ prio
$ python3 examples/swig-wrapper/main.py
```

[Binary releases](https://packaging.python.org/guides/packaging-binary-extensions/#publishing-binary-extensions) are discouraged when there are C-extensions because they require compatible versions of libraries (like glibc, etc). We can't use [PEP 513](https://www.python.org/dev/peps/pep-0513/) and the `manylinux1` tag because of the dependencies on `nss` and `nspr`.

I'll probably create a separate docker container with Ubuntu 16.04 to test that everything works as expected. This should also work on EMR (but requires touching `bootstrap.sh`...).

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -10,6 +10,9 @@ prio = {path = ".", editable = true}
 pytest = "*"
 coverage = "*"
 pytest-cov = "*"
+setuptools = "*"
+wheel = "*"
```
Comment (acmiyaguchi) [inline Pipfile]: This should be removed

File: examples/swig-wrapper/main.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from prio.lib import prio
+from prio import libprio as prio
```
Comment (acmiyaguchi) [inline examples/swig-wrapper/main.py]: It might be worth rewriting the entire example to use `libprio` directly.

File: Pipfile (hunk 1 above)
Comment (sunahsuh) [reply Pipfile]: Can you expand on this comment? Why is it included here now and when should it be removed?

Comment (acmiyaguchi) [reply Pipfile]: I originally planned to build the package as a wheel, but pypi doesn't allow uploading packages with the linux tag that `bdist_wheel` produces in the docker image. I'll remove it in a second pass of this PR where I'll upload the package to the real pypi at the end.

Comment (acmiyaguchi) [inline Pipfile]:
```suggestion
```

**Comment by acmiyaguchi** [comment]
The package name is actually still `prio`, but the wrapper is named libprio. I filed an issue in https://github.com/mozilla/libprio/issues/56 with the hope of moving the swig wrapper close to the source.

**Pull request merged**